lib/remote-proxy.js

```diff
@@ -19,17 +19,31 @@ class RemoteProxy extends EventEmitter {
     methods.filter(method => this[method] === undefined)
       .forEach((method) => {
         this[method] = (...args) => {
-          let callback = args[args.length - 1];
+          let resend = args[args.length - 1];
+          let callback = args[args.length - 2];
 
-          if (typeof(callback) === 'function') {
-            args = Array.prototype.slice.call(args, 0, -1);
+          if (typeof callback === 'function' && typeof resend === 'boolean') {
+            args = args.slice(0, -2);
+          } else if (typeof resend === 'function') {
+            callback = resend;
+            resend = false;
+            args = args.slice(0, -1);
           } else {
             callback = null;
+            resend = false;
+          }
+
+          if (resend) {
+            this._connection.callMethodWithResend(
+              this._interfaceName, method, args, callback
+            );
+            return;
           }
 
           this._connection.callMethod(
             this._interfaceName, method, args, callback
           );
+
```

Comment:
Is this blank line needed?
Also, I'd suggest to just use if\else on the line above as it should look better imo.

Reply:
@lundibundi, if/else will increase indentation in the second block, I don't think that will look better.

Reply:
I meant conceptually it will present the actual if/else (as that's what's happening here) and avoid a return hack for indentation purposes only. I'd understand if it affected the whole/almost whole body of a function but here it's just one statement.